web/src/engine/websites/Tapas.ts

```diff
@@ -0,0 +1,85 @@
+import { Tags } from '../Tags';
+import icon from './Tapas.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON } from '../platform/FetchProvider';
+
+type APIMangas = {
+    data: {
+        items: APIManga[]
+    }
+}
+
+type APISingleManga = {
+    data: {
+        id: number,
+        title: string
+    }
+}
+
+type APIManga = {
+    seriesId: number,
+    title: string
+}
+
+type APIChapters = {
+    data: {
+        body: string;
+    }
+}
+
+@Common.PagesSinglePageCSS('article.viewer__body img.content__img')
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = 'https://story-api.tapas.io/cosmos/api/v1/landing/';
+
+    public constructor() {
+        super('tapas', `Tapas`, 'https://tapas.io', Tags.Media.Manga, Tags.Language.English, Tags.Source.Official, Tags.Accessibility.RegionLocked);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/series/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const seriesId = (await FetchCSS<HTMLMetaElement>(new Request(url), 'meta[property="al:android:url"]'))[0].content.split('/').pop();
+        const { data } = await FetchJSON<APISingleManga>(new Request(new URL(`/series/${seriesId}?`, this.URI), {
```

Comment:
Request uses `URI` instead of `apiUrl`

Reply:
There is nothing wrong. Website uses both.